src/app/zap-templates/zcl/data-model/chip/service-area-cluster.xml

```diff
@@ -0,0 +1,243 @@
+
+  <struct name="ProgStruct">
+    <cluster code="0x0150"/>
+    <item name="LocationId"             type="int32u"        optional="false"/>
+    <item name="Status"                 type="OpStatusEnum"  optional="false"/>
+    <item name="TotalOperationalTime"   type="elapsed_s"     optional="true" isNullable="true"/>
+    <item name="EstimatedTime"          type="elapsed_s"     optional="true" isNullable="true"/>
+  </struct>
+
+
+
+  <!-- Enum definitions for select locations command response-->
+  <enum name="SelectLocationsStatus" type="enum8">
+    <cluster code="0x0150"/>
+    <item value="0x00" name="Success"/>
+    <item value="0x01" name="UnsupportedLocation"/>
+    <item value="0x02" name="DuplicatedLocations"/>
+    <item value="0x03" name="InvalidInMode"/>
+    <item value="0x04" name="InvalidSet"/>
+  </enum>
+
+  <!-- Enum definitions for skip location command response-->
+  <enum name="SkipCurrentStatus" type="enum8">
+    <cluster code="0x0150"/>
+    <item value="0x00" name="Success"/>
+    <item value="0x01" name="InvalidLocation"/>
+    <item value="0x02" name="InvalidInMode"/>
+  </enum>
+
+
+  <cluster>
+    <domain>General</domain>
+    <name>Service Area</name>
+      <code>0x0150</code>
+    <define>SERVICE_AREA_CLUSTER</define>
+    <description>The Service Area cluster provides an interface for controlling the locations where a device should operate, and for querying the current location.</description>
+
+    <!-- Attributes -->
+    <attribute side="server" code="0x0000" define="SupportedLocations"  type="array"     entryType="LocStruct"  writable="false" isNullable="false" optional="false">SupportedLocations</attribute>
+    <attribute side="server" code="0x0001" define="SupportedMaps"       type="array"     entryType="MapStruct"  writable="false" isNullable="false" optional="false">SupportedMaps</attribute>
+    <attribute side="server" code="0x0002" define="SelectedLocations"   type="array"     entryType="int32u"     writable="false" isNullable="true"  optional="false" scene="true">SelectedLocations</attribute>
+    <attribute side="server" code="0x0003" define="CurrentLocation"     type="int32u"                           writable="false" isNullable="true"  optional="true">CurrentLocation</attribute>
+    <attribute side="server" code="0x0004" define="EstimatedEndTime"    type="epoch_s"                          writable="false" isNullable="true"  optional="true" >EstimatedEndTime</attribute>
+    <attribute side="server" code="0x0005" define="Progress"            type="array"     entryType="ProgStruct" writable="false" isNullable="true"  optional="true">Progress</attribute>
+
+    <!-- Commands -->
+    <command source="client" code="0x00" name="SelectLocations" response="SelectLocationsResponse" optional="false">
+      <description>
+        Command used to select a set of device locations, where the device is to operate
+      </description>
+      <arg name="NewLocations" type="int32u" array="true"/>
```

Comment:
Should the type be `type="array" entryType="int32u"`?

Should this attribute be nullable `isNullable="true"`?